chia/wallet/trade_manager.py

```diff
@@ -488,11 +490,15 @@ async def _create_offer_for_ids(
                     else:
                         asset_id = id
                         wallet = await self.wallet_state_manager.get_wallet_for_asset_id(asset_id.hex())
-                    if not callable(getattr(wallet, "get_coins_to_offer", None)):  # ATTENTION: new wallets
-                        raise ValueError(f"Cannot offer coins from wallet id {wallet.id()}")
-                    coins_to_offer[id] = await wallet.get_coins_to_offer(
-                        asset_id, uint64(abs(amount)), min_coin_amount, max_coin_amount
-                    )
+
+                    if isinstance(wallet, (CATWallet, Wallet)):
+                        coins_to_offer[id] = list(
+                            await wallet.get_coins_to_offer(uint64(abs(amount)), min_coin_amount, max_coin_amount)
+                        )
+                    elif asset_id is not None and isinstance(wallet, (NFTWallet, DataLayerWallet)):
```

Comment:
Maybe the `asset_id` check should be done inside the `if` and provide its own error message?  If we just don't have an asset id, that doesn't mean an NFT wallet is incapable of offers.  Also, I thought asset ID was a CAT term and this is for not-CAT wallets.